Tasks/Common/MSBuildHelpers/ArgumentFunctions.ps1

```diff
@@ -20,8 +20,39 @@ function Format-MSBuildArguments {
             $MSBuildArguments = "$MSBuildArguments /p:VisualStudioVersion=`"$VSVersion`""
         }
 
+        $userAgent = Get-UserAgentString
+        if (!([string]::IsNullOrEmpty($userAgent))) {
+            $MSBuildArguments = "$MSBuildArguments /p:_MSDeployUserAgent=`"$userAgent`""
```

Comment:
Is this actually applicable to the MSBuild task, or is this only applicable to the VSBuild task? The exact command line is printed as `##command` so it stands out in the logs.

Is there anything in the user agent string that needs to esacped. It looks like most of the parts are IDs and probably are integers. Although, I don't know about EnvironmentId - is it an integer? Any other strings in here that may need to be escaped?